lib/auth/Vault.ts

```diff
@@ -63,24 +63,38 @@ export function vaultSignatureCb(
         accountQuota: info.accountQuota || {},
     });
 }
+export type AuthV2RequestParams = {
+    version: 2;
+    log: RequestLogger;
+    data: {
+        accessKey: string;
+        algo: string;
+        authType: 'query' | 'header' | 'REST-HEADER' | 'REST-QUERY-STRING';
+        securityToken: string;
+        signatureAge?: number;
+        signatureFromRequest: string;
+        signatureVersion: string;
+        stringToSign: string;
+    };
+};
 
 export type AuthV4RequestParams = {
     version: 4;
     log: RequestLogger;
     data: {
         accessKey: string;
-        signatureFromRequest: string;
+        algo?: string;
+        authType?: 'query' | 'header' | 'REST-HEADER' | 'REST-QUERY-STRING';
+        credentialScope?: string;
```

Comment:
can we type `credentialScope`, with the possible values?

Reply:
We cannot, this is dynamically extracted from the string to sign.